pkg/openapi.go

```diff
@@ -1036,6 +1036,11 @@ func (ctx *resourceContext) genPropertySpec(propName string, p openapi3.SchemaRe
 		propertySpec.Default = p.Value.Default
 	}
 
+	// Is this property marked as a secret?
+	if isSecret, ok := p.Value.Extensions["x-pulumi-secret"]; ok {
```

Comment:
Can you please make the `x-pulumi-secret` an exported constant somewhere? Perhaps, an `extensions.go` file? That way, providers can access that if they have to modify their OpenAPI spec before passing it to `pulschema`.